sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3497,24 +3500,25 @@ static void cleanupAfterEvent(pi_event Event) {
   // any of the Event's data members that need to be read/reset as
   // part of the cleanup operations.
   auto Queue = Event->Queue;
+  if (Queue) {
```

Comment:
We needed to guard the code by adding "if (Queue)" because when the event is released we don't need to reuse the fence and command list associated with queue (that is being deleted)